### ISSUE TITLE
Introduce a Dataset Representation

### ISSUE THREAD

**Issue by eFroD** [PeePeeJay/racoons#21]
We need a data representation to feed it into our modules.

Requirements for a dataset within the project are:

- [x] store the raw dataframe  
- [x] assign the datatype of the columns 
- [x] create features
- [x] create targets

**Comment by PeePeeJay** [comment]
```mermaid
classDiagram
    %%Data <|-- Classification
    %%Model <|-- Regression
    
    class DataSet{
      + _raw_dataframe
      + dataframe
      + features
      + targets 
      + scale_levels   
      + _read_dataframe(features, targets, scale_levels)
    }
```

**Comment by eFroD** [comment]
Merged in #31.
